client/src/components/UploadCodeDailog.js

```diff
@@ -0,0 +1,252 @@
+import React, { useState, useMemo } from "react";
+import Button from "@material-ui/core/Button";
+import { Dialog, makeStyles, TextField, Typography } from "@material-ui/core";
+import CodeMirror from "@uiw/react-codemirror";
+import "codemirror/keymap/sublime";
+import "codemirror/theme/monokai.css";
+
+//modes ==> tells the codemirror library about the kind of
+//file we are opening so that respective syntax highlighting can be achieved.
+
+const modes = [
```

Comment:
can we extract this into a constants file - as we will probably be using this in multiple places

Reply:
Done as suggested